overlord/snapstate/check_snap.go

```diff
@@ -48,9 +51,131 @@ var featureSet = map[string]bool{
+// but snapd can add more as needed).
+//
+// To avoid [ug]id overlaps, snapd shall only assign [ug]ids >= 524288
+// (0x00080000) and <= 983040 (0x000F0000, ie the first 65536 range under LXD's
+// minimum where the lower 16 bits are all zeroes). While [ug]ids >= 1001062400
+// (0x3BAB0000, the first 65536 range above LXD's maximum where the lower 16
+// bits are all zeroes) would also avoid overlap, considering nested containers
+// (eg, LXD snap runs a container that runs a container that runs snapd),
+// choosing >= 1001062400 would mean that the admin would need to increase the
+// LXD id range for these containers for snapd to be allowed to create its
+// [ug]ids in the deeply nested containers. The requirements would both be an
+// administrative burden and artificially limit the number of deeply nested
+// containers the host could have.
+//
+// Looking at the LSB and distribution defaults for login.defs, we can observe
+// uids and gids in the system's initial 65536 range (ie, 0-65536):
+//
+// - 0-99        LSB-suggested statically assigned range (eg, root, daemon,
+//               etc)
+// - 0           mandatory 'root' user
+// - 100-499     LSB-suggested dynamically assigned range for system users
+//               (distributions often prefer a higher range, see below)
+// - 500-999     typical distribution default for dynamically assigned range
+//               for system users (some distributions use a smaller
+//               SYS_[GU]ID_MIN)
+// - 1000-60000  typical distribution default for dynamically assigned range
+//               for regular users
+// - 65535 (-1)  should not be assigned since '-1' might be evaluated as this
+//               with set[ug]id* and chown families of functions
+// - 65534 (-2)  nobody/nogroup user for NFS/etc [ug]id anonymous squashing
+// - 65519-65533 systemd recommended reserved range for site-local anonymous
+//               additions, etc
+//
+// To facilitate potential future use cases within the 65536 range snapd will
+// assign from, snapd will only assign from the following subset of ranges
+// relative to the range minimum (ie, its 'base' which has the lower 16 bits
+// all set to zero):
+//
+// - 60500-60999 'scope: shared' system-usernames
+// - 61000-65519 'scope: private' system-usernames
+//
+// Since the first [ug]id range must be >= 524288 and <= 983040 (see above) and
+// following the above guide for system-usernames [ug]ids within this 65536
+// range, the lowest 'scope: shared' user in this range is 584788 (0x0008EC54).
+//
+// Since this number is within systemd-nspawn's range of 524288-1879048191
+// (0x00080000-0x6FFFFFFF), the number's lower 16 bits are not all zeroes so
+// systemd-nspawn won't detect this allocation and could potentially assign the
+// 65536 range starting at 0x00080000 to a container. snapd will therefore also
+// create the 'snap-range-524288-root' user and group with [ug]id 524288 to
```

Comment:
why the -root suffix ?    I wonder if these should use the hex representation  snap-range-0x080000-root in their name, it would seem a bit easier to parse

Reply:
@pedronis - I chose '-root' because this would correspond to the '0' uid/gid of the 65536 range, which we would want to handle carefully if we changed how we use the range, etc and this is a reminder of what it would map to.

Reply:
I see it's the translated id of root for the range

Reply:
I think the prefix for this username should be snapd-  not snap-